### ISSUE TITLE
wildcard_in_or_patterns doesn't ignore #[nonexhaustive]

### ISSUE THREAD

**Issue by andrewbanchich** [rust-lang/rust-clippy#13350]
### Summary

Some types are annotated with `#[nonexhaustive]`, forcing a wildcard pattern. This lint conflicts with that, which then forces us to use an `#[allow(clippy::wildcard_in_or_patterns)]`.

It would be great if this lint took `nonexhaustive` into account so that it doesn't trigger for wildcard patterns as long as all other patterns were exhaustively matched upon.

### Version

```text
release: 1.80.1
```
